### PR TITLE
Example generator

### PR DESCRIPTION
**Pull request by Winetq**

### CODE REVIEW

File: generator/TemperatureSensor.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+            int divisor = Int32.Parse(ConfigurationManager.AppSettings["HOW_MANY_DATA_PER_MINUTE_FOR_TEMPERATURE"]);
+            this.timeout = numerator / divisor;
+            this.channel = channel;
+            this.channel.QueueDeclare(queueName, false, false, false, null);
+        }
+
+        public void publish() {
+            int x = random.Next(min, max);
+            System.Threading.Thread.Sleep(timeout);
+            channel.BasicPublish("", queueName, null, BitConverter.GetBytes(x));
```
Comment (Michal299) [inline generator/TemperatureSensor.cs:41]: Please add log message with the `x` value before publishing it. 
```suggestion
            Console.WriteLine($"Publishing generated value: {x}"}
            channel.BasicPublish("", queueName, null, BitConverter.GetBytes(x));
```

File: generator/App.config [hunk 2]
```diff
@@ -0,0 +1,9 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
+<configuration>
+	<appSettings>
+		<add key="MIN_TEMPERATURE" value="0" />
+		<add key="MAX_TEMPERATURE" value="100" />
+		<add key="HOW_MANY_DATA_PER_MINUTE_FOR_TEMPERATURE" value="12" />
+		<add key="MINUTE_IN_MILISECONDS" value="60000" />
```
Comment (Michal299) [inline generator/App.config]: NIT, you can use: 
```
(int)TimeSpan.FromMinutes(1).TotalMilliseconds;
```

File: generator/App.config [hunk 3]
```diff
@@ -0,0 +1,9 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
+<configuration>
+	<appSettings>
+		<add key="MIN_TEMPERATURE" value="0" />
+		<add key="MAX_TEMPERATURE" value="100" />
+		<add key="HOW_MANY_DATA_PER_MINUTE_FOR_TEMPERATURE" value="12" />
```
Comment (Michal299) [inline generator/App.config]: WDYT about smaller value? It seems a little bit talkative for me, 3-5 times per minute?

File: generator/TemperatureSensor.cs [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        private IModel channel;
+        private int min;
+        private int max;
+        private int timeout;
+
+        public TemperatureSensor(IModel channel) 
+        {
+            this.random = new Random();
+            this.queueName = "temperature_sensor";
+            this.min = Int32.Parse(ConfigurationManager.AppSettings["MIN_TEMPERATURE"]);
```
Comment (Michal299) [inline generator/TemperatureSensor.cs]: IDK how it works with console application, but you can use appsettings.json (at least in webapi project type). With this you can avoid neccesity of mapping values by using just:
```
_config.GetValue<String>("RabbitHost");
_config.GetValue<Int32>("RabbitPort")
```
when the config file is just simple json:
```
{
  "Logging": {
    "LogLevel": {
      "Default": "Information",
      "Microsoft.AspNetCore": "Warning"
    }
  },
  "AllowedHosts": "*",
  "RabbitUsername": "admin",
  "RabbitPassword": "admin",
  "RabbitPort": 5672,
  "RabbitHost": "rabbitmq",
  "QueueNameTemperature": "temperature_sensor"
}
```
For more details you can see: https://dotnettutorials.net/lesson/asp-net-core-appsettings-json-file/

File: generator/TemperatureSensor.cs (hunk 1 above)
Comment (Winetq) [reply generator/TemperatureSensor.cs:41]: Done.

File: generator/TemperatureSensor.cs (hunk 4 above)
Comment (Winetq) [reply generator/TemperatureSensor.cs]: appsettings.json also works with console applications so I used it

**Review by Michal299 — APPROVED** [review]

**Pull request merged**